js/src/personal/riyuan/src/navigation/Navigation.tsx

```diff
@@ -0,0 +1,145 @@
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core'
+import pfpImage from '../../assets/images/pfp.jpg'
+import styles from './Navigation.module.css'
+
+interface CustomHeaderProps {
+  data: {
+    image: string
+    name: string
+  }
+  toggleShell: () => void
+  isClosed: boolean
+}
+
+const CustomHeader = ({ data, toggleShell, isClosed }: CustomHeaderProps) => (
+  <header>
+    <div className={styles.ImageText}>
+      <span className={styles.image}>
+        <Image
+          src={data.image}
+          alt={'Profile pic'}
+        />
+      </span>
+      {isClosed && (
+        <div className={`${styles.text} ${styles.logoText}`}>
+          <Text className={styles.text}>{`${data.name}'s`}</Text>
+          <Text className={styles.text}>{'Web'}</Text>
+        </div>
+      )}
+    </div>
+    <Button
+      className={styles.toggle}
+      onClick={toggleShell}
+    >
+      <FontAwesomeIcon icon={faCircle} />
+    </Button>
+  </header>
+)
+
+const NavItem = ({
+  title,
+  icon,
+  isClosed,
+}: {
+  title: string
+  icon: IconDefinition
+  isClosed: boolean
+}) => (
+  <div className={styles.navLink}>
+    <Link
+      to={`#${title}`}
+      className={styles.navItem}
+    >
+      <FontAwesomeIcon
+        icon={icon}
+        className={styles.icon}
+      />
+      {isClosed && <span className={`${styles.text}`}>{title}</span>}
+    </Link>
+  </div>
+)
+const NavList = ({
+  items,
+  isClosed,
+}: {
+  items: { title: string; icon: IconDefinition }[]
+  isClosed: boolean
+}) => (
+  <Stack className={styles.navList}>
+    {items.map((item, index) => (
+      <NavItem
+        key={index}
+        title={item.title}
+        icon={item.icon}
+        isClosed={isClosed}
+      />
+    ))}
+  </Stack>
+)
+
+const SearchBox: React.FC = () => (
+  <List className={styles.searchBox}>
+    <div className={styles.icon}>
+      <FontAwesomeIcon icon={faMagnifyingGlass} />
+    </div>
+    <Input
+      type={'text'}
+      placeholder={'Search..'}
+    />
+  </List>
+)
+
+const MenuBar = ({
+  items,
+  isClosed,
+}: {
+  items: { title: string; icon: IconDefinition }[]
+  isClosed: boolean
+}) => (
+  <div className={styles.MenuBar}>
+    <div className={styles.Menu}>
+      <SearchBox />
+      <NavList
+        items={items}
+        isClosed={isClosed}
+      />
+    </div>
+  </div>
+)
+
+export function Nav() {
+  const headerData = {
+    name: 'Riyuan',
+    image: pfpImage,
+  }
+  const navItems = [{ title: 'Home', icon: faHouse }]
+  const [isClosed, setIsClosed] = useState(true)
+  const toggleShell = () => {
+    setIsClosed(!isClosed)
+  }
+
+  return (
+    <nav className={`${styles.nav} ${styles.dark}`}>
+      <div className={`${styles.shell} ${isClosed ? styles.close : ''}`}>
+        <CustomHeader
+          data={headerData}
+          toggleShell={toggleShell}
+          isClosed={!isClosed}
+        />
+        <MenuBar
+          items={navItems}
+          isClosed={!isClosed}
```

Comment:
can you clarify `isClosed`? It seems we want to name it isOpen instead since usages above are like `isClosed && <text when open>` and we are negating it here before passing into child components.